### PR TITLE
Add filtration support

### PR DESCRIPTION
**Pull request by reds-heig**
#### Reference Issues/PRs

#16 

#### What does this implement/fix? Explain your changes.

* Implements support for the different filtrations supported by `flagser`
* Fix disabling of std::cout which was disabled until the programs ends :cry: 
* Clean setup.py unnecessary  installation of pybind11

#### Any other comments?

Add check of filtration if both bindings and python code for available filtrations.

### CODE REVIEW

File: pyflagser/flagser.py [hunk 1]
```diff
@@ -41,6 +41,17 @@ def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
         often ``100,000``. Increase for higher precision, decrease for faster
         computation. A negative value computes highest possible precision.
 
+    filtration : string, optional, default: ``max``
+        filtration algorithm use the specified algorithm to compute the
```
Comment (ulupo) [inline pyflagser/flagser.py]: @MonkeyBreaker could you fix the wording of the first sentence?

Comment (MonkeyBreaker) [reply pyflagser/flagser.py]: Is `Use the specified algorithm to compute the filtration.` better ?

File: pyflagser/flagser.py [hunk 2]
```diff
@@ -41,6 +41,17 @@ def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
         often ``100,000``. Increase for higher precision, decrease for faster
         computation. A negative value computes highest possible precision.
 
+    filtration : string, optional, default: ``max``
```
Comment (ulupo) [inline pyflagser/flagser.py]: Since this is a string, use ``'max'``

File: pyflagser/flagser.py (hunk 1 above)
Comment (ulupo) [reply pyflagser/flagser.py]: I was thinking just "Algorithm determining the filtration."

File: pyflagser/flagser.py [hunk 3]
```diff
@@ -41,6 +41,17 @@ def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
...
+        filtration algorithm use the specified algorithm to compute the
+        filtration. Warning: if an edge filtration is specified, it is assumed
+        that the resulting filtration is consistent, meaning that the
+        filtration value of every simplex of dimension at least two should
+        evaluate to a value that is at least the maximal value of the
+        filtration values of its containing edges. For performance reasons,
+        this is not checked automatically. Possible values are = ['dimension',
```
Comment (ulupo) [inline pyflagser/flagser.py]: replace "are =" with "are: "

**Comment by MonkeyBreaker** [comment]
```
filtration : string, optional, default: `max`
        Algorithm dtermining the filtration. Warning: if an edge filtration is
        specified, it is assumed that the resulting filtration is consistent,
        meaning that the filtration value of every simplex of dimension at
        least two should evaluate to a value that is at least the maximal value
        of the filtration values of its containing edges. For performance
        reasons, this is not checked automatically.  Possible values are :
            ['dimension', 'zero', 'max', 'max3', 'max_plus_one', 'product',
            'sum', 'pmean', 'pmoment', 'remove_edges', 'vertex_degree']
```

Is it better this way ?

**Comment by ulupo** [comment]
@MonkeyBreaker typo in "dtermining", also what I meant is to write "``'max'``"

**Comment by MonkeyBreaker** [comment]
Sorry for the type, i edited previous message

**Comment by ulupo** [comment]
And take care of the linting after "are: ".

**Comment by ulupo** [comment]
Double ticks, please. Single ticks are displayed differently by sphinx (they make things look italic).

File: pyflagser/flagser.py [hunk 4]
```diff
@@ -41,6 +41,16 @@ def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
         often ``100,000``. Increase for higher precision, decrease for faster
         computation. A negative value computes highest possible precision.
 
+    filtration : string, optional, default: `'max'`
```
Comment (ulupo) [inline pyflagser/flagser.py]: Double ticks, `` instead of `

File: pyflagser/flagser.py [hunk 5]
```diff
@@ -88,8 +98,13 @@ def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
     else:
         _max_dimension = max_dimension
 
+    if filtration not in implemented_filtrations:
+        print('unrecognized {}, using max'.format(filtration))
```
Comment (ulupo) [inline pyflagser/flagser.py]: Capital U in unrecognized

File: pyflagser/flagser.py [hunk 6]
```diff
@@ -41,6 +41,16 @@ def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
         often ``100,000``. Increase for higher precision, decrease for faster
         computation. A negative value computes highest possible precision.
 
+    filtration : string, optional, default: `"max"`
```
Comment (ulupo) [inline pyflagser/flagser.py]: Still not right. I've fixed it myself as it is difficult to explain here :)

Comment (MonkeyBreaker) [reply pyflagser/flagser.py]: Thank you and sorry, not quite familiar with sphinx

**Pull request merged**